src/util/generate-token.js

```diff
@@ -0,0 +1,35 @@
+import { installTimeStorageKey } from 'src/imports/background'
+
+const API_PATH = '/user-token'
+
+export const USER_ID = 'user-id'
+
+export default async function generateTokenIfNot(installTime) {
+    const userId = (await browser.storage.local.get(USER_ID))[USER_ID]
+
+    // if install time is not present then current time stores as install time and send to server same
+    if (!installTime) {
+        installTime = Date.now()
+        browser.storage.local.set({ [installTimeStorageKey]: installTime })
+    }
+
+    const host =
+        'https://a8495szyaa.execute-api.eu-central-1.amazonaws.com/' +
```

Comment:
I would have this value assigned to a constant and maybe allow this to be specified in the `generateTokenIfNot` fn params. You could just assign the default value of the param to be the constant. I presume we'll assign the API endpoint a proper name sometime.

i.e.
```js
generateTokenIfNot({ installTime = Date.now(), hostname = API_HOST, pathname = API_PATH }) {...}
```

You could still have the `process.env` check where you assign the `API_HOST` constant.
I would also either give `installTime` a default value here or throw an Error if it's not defined, as when you call it in `onInstall`/`onUpdate` there is an easy chance for it not to be defined.

Reply:
Yes that makes sense :)